File: Casks/zcode.rb
cask "zcode" do
  arch arm: "arm64", intel: "x64"

  version "3.11.2"
  # 上游按架构分包（core 的 zcode cask 给的只是 macos-arm64 版）。
  # 镜像到本仓 Release（tag zcode-<ver>，双资产名沿用上游
  # ZCode-<ver>-mac-<arch>.dmg）：本 tap 政策是所有 cask 都镜像最新版；
  # url 三段都用 #{version}/#{arch} 双插值——版本升级时 url 自动跟随，
  # audit 的 "Use `sha256 :no_check` when URL is unversioned" 也不会触发。
  # 有更新时 updater/zcode.swift 下载双包、上传 Release、改写 sha 行
  # （url 行插值已覆盖新版本，不动）。
  # 注：dmg 内含 Applications -> /Applications 符号链接，公式的 DmgUnpackStrategy
  # 会崩（AGENTS 11.8），cask 走自己的挂载流程不受影响——这也是它做成 cask 的原因。
  sha256 arm:   "cfa43b90ec74732ee3ee1262803d775658a3c954cc4cc0a9a1bec0f9c6dcbf98",
         intel: "12cf306271a6bfb5f4100b9c735e07a22912051d27af845f98064495e41fd736"

  url "https://github.com/Bemly/homebrew-tahoe/releases/download/zcode-#{version}/ZCode-#{version}-mac-#{arch}.dmg"
  name "ZCode"
  desc "AI-assisted development environment"
  homepage "https://zcode.z.ai/en/"

  auto_updates true
  # 本 tap 只收录 Intel(x86_64) + macOS 26(Tahoe) 及以上可用的二进制；
  # arm 段为 Apple Silicon 同包顺带覆盖（universal 时代前的分包形态）。
  depends_on macos: :tahoe

  app "ZCode.app"

  uninstall quit: "dev.zcode.app"

  zap trash: [
        "~/Library/Application Support/com.apple.sharedfilelist/com.apple.LSSharedFileList.ApplicationRecentDocuments/dev.zcode.app.sfl*",
        "~/Library/Application Support/ZCode",
        "~/Library/Caches/@zcodedesktop-updater",
        "~/Library/Preferences/dev.zcode.app.plist",
        "~/Library/Services/Open in ZCode.workflow",
      ],
      rmdir: "~/.zcode"
end

File: updater/zcode.swift
// updater/zcode.swift —— 与 Casks/zcode.rb 一一同名的检查器入口
//
// ZCode（z.ai 的 AI 辅助开发环境，Electron 桌面 app）在 homebrew/core 里是 **cask**，
// core 那份是 macos-arm64 的 dmg；本 tap 取上游 CDN 的双架构 dmg（x64 + arm64）。
//
// 版本判据：formulae.brew.sh 的 cask API（api/cask/zcode.json 的 .version），
// 走 UpdaterCore 的 brewCask 分支——不是公式的 .versions.stable。
// 双架构产物（archArtifacts + downloadURLForArch，走 runDualArchCheck）：
// 有更新时两个包都下载实算 sha，改写 cask 的 version 行 + 双 sha 行；
// url 行不动（#{version}/#{arch} 插值已覆盖新版本）。
// 两处必须自己给，不能沿用 core 的值：
//   - 直链：core 的 url 是 arm64 的，必须由 downloadURLForArch 模板拼出双直链；
//   - sha256：core 那份属于 arm64 包，对 x64 无效 → 双双实算。
//
// 直引上游 CDN、链接规律（路径与文件名都随版本变化）→ uploadRelease 保持 false：
// 不上传本仓 Release，也不把 url 改写为 Release 地址（与 workbuddy/doubao-ime 不同）。
// cask 无 bottle 机制，watcher 更新 cask 后不触发 bottle.yml。
//
// 注意：Swift 要求实参顺序与 CheckConfig.init 的形参声明一致。

import Foundation

@main
struct ZcodeCheck {
    static func main() {
        runCheck(CheckConfig(
            formula: "zcode",
            formulaPath: "Casks",
            isCask: true,
            brewName: "zcode",
            brewCask: true,
            archArtifacts: ["arm64", "x64"],
            downloadURLForArch: { version, arch in
                "https://cdn-zcode.z.ai/zcode/electron/releases/\(version)/macos-\(arch)/ZCode-\(version)-mac-\(arch).dmg"
            }
        ))
    }
}
